crates/matrix-sdk-base/src/rooms/normal.rs

```diff
@@ -207,8 +208,21 @@ impl Room {
     }
 
     /// Is this room considered a direct message.
-    pub fn is_direct(&self) -> bool {
-        !self.inner.read().unwrap().base_info.dm_targets.is_empty()
+    pub async fn is_direct(&self) -> Option<bool> {
+        match self.state() {
+            RoomState::Joined | RoomState::Left => {
+                Some(!self.inner.read().unwrap().base_info.dm_targets.is_empty())
+            }
+            RoomState::Invited => {
+                let get_member_result = self.get_member(self.own_user_id()).await;
+                let member = get_member_result.ok()??;
```

Comment:
I don't think we should just swallow errors from `get_member` here. Instead, I think it would make sense to:

- Update the function signature to `Result<Option<bool>>`
- Short-circuit with `?` if there is an error from `get_member`
- Log an event (info level seems appropriate) if `get_member` returns `None`, and return `Ok(None)`

We could also consider treating `None` as `false` in this function, rather than in higher-level functions. I'm not sure whether there is a semantic difference between `"is_direct": false` and not specifying it in the member event anyways.

Reply:
Applied [here](https://github.com/matrix-org/matrix-rust-sdk/pull/1775/commits/d3ccf7ca6d11850c84620685f23d7669c9da272e)
Wdyt?